### PR TITLE
Fix .zip extraction

### PR DESCRIPTION
**Pull request by ngosang**
This commit fix .zip extraction used in the mods store. The old code only works for some .zips. Test it with Nature and Food mods.

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -862,15 +862,18 @@ int ModApiMainMenu::l_extract_zip(lua_State *L)
...
+			std::string fullpath_dir = fs::RemoveLastPathComponent(fullpath);
 
-			if (files_in_zip->isDirectory(i)) {
-				if (! fs::CreateAllDirs(fullpath) ) {
-					fs->removeFileArchive(fs->getFileArchiveCount()-1);
-					lua_pushboolean(L,false);
-					return 1;
+			if (! files_in_zip->isDirectory(i)) {
+				
+				if (! fs::PathExists(fullpath_dir)) {
```
Comment (ShadowNinja) [inline src/script/lua_api/l_mainmenu.cpp]: The `!` shouldn't be separated by a space and there shouldn't be an extra newline.
You should also combine the second conditional with the one below.

Comment (ngosang) [inline src/script/lua_api/l_mainmenu.cpp]: changed.

**Comment by ngosang** [comment]
@nerzhul This is simple to review.

**Comment by nerzhul** [comment]
Fix is correct.
Please merge your commits by rebasing them (git rebase -i HEAD~3) and fix whitespaces.
I'll merge it after

**Comment by ngosang** [comment]
Done. I can't see the whitespaces to fix.

**Comment by nerzhul** [comment]
Commited, thanks
Commit: e102cbd840c3dc3d52865868369aea3b8ee21028

**Pull request closed without merging**